src/app/interaction-model/OperationTypes.h

```diff
@@ -0,0 +1,90 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <access/SubjectDescriptor.h>
+#include <app/ConcreteAttributePath.h>
+#include <app/ConcreteCommandPath.h>
+#include <lib/core/Optional.h>
+#include <lib/support/BitFlags.h>
+
+#include <cstdint>
+
+namespace chip {
+namespace app {
+namespace InteractionModel {
+
+/// Contains common flags among all interaction model operations: read/write/invoke
+enum class OperationFlags : uint32_t
+{
+    kInternal = 0x0001, // Internal request for data changes (can bypass checks/ACL etc.)
+};
+
+/// This information is available for ALL interactions: read/write/invoke
+struct OperationRequest
+{
+    OperationFlags operationFlags;
+    chip::Access::SubjectDescriptor subjectDescriptor; // Current authentication data
```

Comment:
In the case of an internal read/write/invoke is this actually provided?

Note I don't know too much about `Access::SubjectDescriptor`, but vaguely know that this comes from an exchange context so I just want to make sure that if we are considering the internal calls that we be careful with the comments about stating this is available to all (including internal)? We can make a comment on subjectDescriptor that is is only valid if not internal

Reply:
I think for internal it has no meaning. Will add a comment.

Reply:
Added comment and switched to optional, to at least have this consistently required to be validated.

I still do not like that we have this duality of both flags and descriptor ... if there are better suggestions let me know.